src/reports_updater.rs

```diff
@@ -293,27 +378,76 @@ where
         self.retrieve_and_match_new_reports().map(|signed_reports| {
             signed_reports
                 .into_iter()
-                .filter_map(|matched_report| self.to_ffi_alert(matched_report).ok())
+                .filter_map(|matched_report| self.to_ffi_alerts(matched_report).ok())
+                .flatten()
                 .collect()
         })
     }
 
     // Note: For now we will not create an FFI layer to handle JSON conversions, since it may be possible
     // to use directly the data structures.
-    fn to_ffi_alert(&self, matched_report: MatchedReport) -> Result<Alert, ServicesError> {
-        let report = matched_report.report.clone().verify()?;
+    fn to_ffi_alerts(&self, matched_report: MatchedReport) -> Result<Vec<Alert>, ServicesError> {
+        let exposures = Self::group_in_exposures(matched_report.clone().tcns, 3600);
+
+        exposures
+            .into_iter()
+            .map(|exposure_tcns| self.to_alert(matched_report.report.clone(), exposure_tcns))
+            .collect()
+    }
+
+    fn to_alert(
+        &self,
+        signed_report: SignedReport,
+        exposure: Exposure,
+    ) -> Result<Alert, ServicesError> {
+        let report = signed_report.clone().verify()?;
 
         let public_report = self.memo_mapper.to_report(Memo {
             bytes: report.memo_data().to_vec(),
         });
 
+        let measurements = exposure.measurements();
+
         Ok(Alert {
-            id: format!("{:?}", matched_report.report.sig),
+            id: format!("{:?}", signed_report.sig),
             report: public_report,
-            contact_time: matched_report.contact_time.value,
+            contact_start: measurements.contact_start,
+            contact_end: measurements.contact_end,
+            min_distance: measurements.min_distance,
         })
     }
 
+    fn group_in_exposures(mut tcns: Vec<ObservedTcn>, separator: u64) -> Vec<Exposure> {
```

Comment:
Naming: I prefer threshold or tolerance to separator. Separator is usually a character used to split a string.

Reply:
I agree, threshold "magnitude or intensity that must be _exceeded_" is more accurate.